File: 1038_Binary_Search_Tree_to_Greater_Sum_Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def bstToGst(self, root: TreeNode) -> TreeNode:
        def dfs(root, s):
            if not root:
                return s
            right = dfs(root.right, s)
            root.val += right
            left = dfs(root.left, root.val)
            return left
        dfs(root, 0)
        return root

File: 1038_Binary_Search_Tree_to_Greater_Sum_Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int dfs(TreeNode node, int s){
        if(node==null){
            return s;
        }
        int right = dfs(node.right, s);
        node.val += right;
        int left = dfs(node.left, node.val);
        return left;
    }
    public TreeNode bstToGst(TreeNode root) {
        dfs(root, 0);
        return root;

    }
}